src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -218,122 +247,176 @@ private static Integer randomLocalOffset(Random random) {
+                            case MONTH:
+                                timestamp = Timestamp.forMonth(random.nextInt(9998) + 1, random.nextInt(12) + 1);
+                                break;
+                            case DAY:
+                                timestamp = Timestamp.forDay(random.nextInt(9998) + 1, random.nextInt(12) + 1,
+                                        random.nextInt(28) + 1 // Use max 28 for simplicity. Not including up to 31 is not going to
+                                        // affect the measurement.
+                                );
+                                break;
+                            case MINUTE:
+                                timestamp = Timestamp.forMinute(random.nextInt(9998) + 1, random.nextInt(12) + 1,
+                                        random.nextInt(28) + 1, // Use max 28 for simplicity. Not including up to 31 is not going to
+                                        // affect the measurement.
+                                        random.nextInt(24), random.nextInt(60), randomLocalOffset(random));
+                                break;
+                            case SECOND:
+                                //added
+                                BigDecimal secondValue = value.getDecimalSecond();
+                                int scale = secondValue.scale();
+                                timestamp = Timestamp.forSecond(random.nextInt(9998) + 1, random.nextInt(12) + 1,
+                                        random.nextInt(28) + 1, // Use max 28 for simplicity. Not including up to 31 is not going to
+                                        // affect the measurement.
+                                        random.nextInt(24), random.nextInt(60), randomSecondWithFraction(random,scale), randomLocalOffset(random));
+                                break;
+                            default:
+                                throw new IllegalStateException();
+                        }
+                        writer.writeTimestamp(timestamp);
+                        if (reader.next() == null){
+                            break;
+                        }
+                    }
+                    int currentSize = 7 * count;
+                    size -= currentSize;
                 }
-                Timestamp value = reader.timestampValue();
-                if (i++ < 100) {
-                    System.out.println(value);
+            } else {
+                Random random = new Random();
+                // Target about 100MB of data. Timestamps will average around 7 bytes.
+                Timestamp.Precision[] precisions = Timestamp.Precision.values();
+                for (int i = 0; i < (size / 7); i++) {
+                    Timestamp.Precision precision = precisions[random.nextInt(precisions.length)];
+                    Timestamp timestamp;
+                    int randomScale = random.nextInt(20);
+                    switch (precision) {
```

Comment:
This `swtich` looks like it's duplicated above, so it should be refactored into a method and called both places.